compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -1698,6 +1708,44 @@ private void checkInLangLib(BLangInvocation iExpr, BType varRefType) {
         if (!langLibMethodExists) {
             dlog.error(iExpr.name.pos, DiagnosticCode.UNDEFINED_FUNCTION, iExpr.name.value);
             resultType = symTable.semanticError;
+            return;
+        }
+
+        checkIllegalStorageSizeChangeMethodCall(iExpr, varRefType);
+    }
+
+    private boolean isPureSealed(BType type) {
+        switch(type.tag) {
+            case TypeTags.ARRAY:
+                return isPureSealed((BArrayType) type);
+            case TypeTags.TUPLE:
+                return isPureSealed((BTupleType) type);
+            case TypeTags.UNION:
+                return isPureSealed((BUnionType) type);
+        }
+        return false;
+    }
+
+    private boolean isPureSealed(BArrayType arrayType) {
+        return (arrayType.state == BArrayState.CLOSED_SEALED);
+    }
+
+    private boolean isPureSealed(BTupleType tupleType) {
+        return (tupleType.restType == null);
+    }
+
+    private boolean isPureSealed(BUnionType unionType) {
+        return unionType.getMemberTypes().stream().allMatch(type -> isPureSealed(type));
+    }
+
+    private void checkIllegalStorageSizeChangeMethodCall(BLangInvocation iExpr, BType varRefType) {
+        if (!modifierFunctions.contains(iExpr.name.getValue())) {
```

Comment:
Shall we extract `iExpr.name.getValue()` out to a variable? We seem to be reusing it at L1747 (at times).